pkg/api/validate/dynamic/dynamic.go

```diff
@@ -45,22 +51,14 @@ type ServicePrincipalValidator interface {
 }
 
 // Dynamic validate in the operator context.
-type Dynamic interface {
-	ServicePrincipalValidator
-
-	ValidateVnet(ctx context.Context, location string, subnets []Subnet, additionalCIDRs ...string) error
-	ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster, subnets []Subnet) error
-	ValidateDiskEncryptionSets(ctx context.Context, oc *api.OpenShiftCluster) error
-	ValidateEncryptionAtHost(ctx context.Context, oc *api.OpenShiftCluster) error
-}
```

Comment:
would prefer not to get rid of our interfaces because we can mock them if they're public rather than structs.